packages/presentation/lib/screen/main_my_bloc.dart

```diff
@@ -11,40 +11,41 @@ abstract class MainBloc extends Bloc {
     CheckStateUseCase checkStateUseCase,
     GenerateNumberUseCase generateNumberUseCase,
   ) =>
-      MainBlocImpl(
+      _MainBlocImpl(
         checkStateUseCase,
         generateNumberUseCase,
       );
 
   TextEditingController get editController;
 
-  void generateRandomNumber();
+  void startNewGame();
+
+  void checkNumber();
 }
 
-class MainBlocImpl extends BlocImpl implements MainBloc {
+class _MainBlocImpl extends BlocImpl implements MainBloc {
   int randomNumber = 1;
   int number = 0;
-  int counter = 0;
   final _editController = TextEditingController();
-  final _screenData = MainData.init();
+  var _screenData = MainData.init();
   final CheckStateUseCase _checkStateUseCase;
   final GenerateNumberUseCase _generateNumberUseCase;
 
   @override
   TextEditingController get editController => _editController;
 
-  MainBlocImpl(
+  _MainBlocImpl(
     this._checkStateUseCase,
     this._generateNumberUseCase,
   );
 
   @override
   void initState() {
     super.initState();
-    _updateData();
+    _updateData(data: _screenData);
```

Comment:
no need to pass _screenData to _updateData as u already pass _screenData in handleData

Reply:
Yes, it was on debugging purposes, forgot to delete it back as it was